python/cudf/cudf/core/dtypes.py

```diff
@@ -211,3 +212,32 @@ def __eq__(self, other):
 
     def __repr__(self):
         return f"StructDtype({self.fields})"
+
+
+class DecimalDtype(ExtensionDtype):
+
+    name = "decimal"
+    _metadata = ("precision", "scale")
```

Comment:
What is this `_metadata` attribute used for?

Reply:
It's documented [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.api.extensions.ExtensionDtype.html#pandas.api.extensions.ExtensionDtype).

Essentially, if you have an ExtensionDtype that encapsulates a bunch of attributes, you should list those in the `_metadata` attribute. Then you get things like `__eq__` for free: https://github.com/pandas-dev/pandas/blob/v1.1.4/pandas/core/dtypes/base.py#L96-L124

Reply:
Thanks!